### ISSUE TITLE
no-unused-properties: Misses v-bind inside "style" section (with repro)

### ISSUE THREAD

**Issue by ilyub** [vuejs/eslint-plugin-vue#1789]
**Checklist**

- [+] I have tried restarting my IDE and the issue persists.
- [+] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** ^6.7.2
- **eslint-plugin-vue version:** ^7.0.0
- **Node version:** 14.18
- **Operating System:** Windows 10

**Please show your full configuration:**

See:
https://github.com/ilyub/repro-vue-no-unused-properties/blob/master/.eslintrc.js

```json5
rules: {
    "vue/no-unused-components": "warn",
    "vue/no-unused-properties": "warn",
  },
```

**What did you do?**	

See:
https://github.com/ilyub/repro-vue-no-unused-properties/blob/master/src/components/HelloWorld.vue

```vue
<template>
...
</template>

<script lang="ts">
export default defineComponent({
  name: "HelloWorld",
  props: {
    msg: String,
    background: String,
    otherProp: String,
  },
});
</script>

<style scoped lang="scss">
h3 {
  background: v-bind(background);
}
</style>
```

**What did you expect to happen?**

One error: from otherProp.

**What actually happened?**

Two errors: from otherProp and background.

Note that background is used in "style" section.

IMPORTANT NOTE:
Bug occurs in the presense of "vue/no-unused-components".
If you disable it then you get one error as expected.

**Repository to reproduce this issue**

https://github.com/ilyub/repro-vue-no-unused-properties

**Comment by FloEdelmann** [comment]
This exact issue has been fixed in the latest release: https://github.com/vuejs/eslint-plugin-vue/releases/tag/v8.4.0

Note that the eslint-plugin-vue version you're using (v7.0.0) was released one year ago and many newer versions exist.

**Comment by ilyub** [comment]
I have upgraded eslint-plugin-vue to 8.4.0 in reproduction repository:
https://github.com/ilyub/repro-vue-no-unused-properties/blob/master/package.json
![image](https://user-images.githubusercontent.com/70687432/152488085-7effc867-086d-423f-9097-d192d92ecebd.png)

**It does not fix issue.**

I have checked version inside node_modules folder:
![image](https://user-images.githubusercontent.com/70687432/152488271-085d6893-664f-41d7-bee3-6ec81d471936.png)
It is surely 8.4.0.

The project where I see the issue was upgraded to version 8.4.0 on 27 January and I am reporting issue more than week after.

The only reason why I mentioned v7.0.0 is because I created reproduction repository with "vue create" command and it installed v7.0.0.

**Please, re-open.**

And please, pay attention to my comment about **combination of "vue/no-unused-components" and "vue/no-unused-properties" rules**:
![image](https://user-images.githubusercontent.com/70687432/152489409-60a5759c-0e9d-438e-82a8-85695ad76c5d.png)

**Comment by FloEdelmann** [comment]
@ota-meshi Can xou have a look at this?

**Comment by ota-meshi** [comment]
I tried debugging it.
It seems that the calling order of `'Program:exit'` and `"VElement[parent.type!='VElement']:exit"` AST node listeners differs depending on the rule registration order. This causes the rule to finish parsing before `v-bind()` processed, which can lead to false positives.
